app/src/main/java/org/simple/clinic/newentry/PatientEntryScreen.kt

```diff
@@ -477,6 +486,35 @@ class PatientEntryScreen(context: Context, attrs: AttributeSet) : RelativeLayout
     }
   }
 
+  private fun prefillAlternateId(alternateId: Identifier) {
+    when (alternateId.type) {
+      Identifier.IdentifierType.BangladeshNationalId -> setAlternateIdTextField(alternateId)
+      Identifier.IdentifierType.EthiopiaMedicalRecordNumber -> setAlternateIdTextField(alternateId)
+      Identifier.IdentifierType.IndiaNationalHealthId -> setAlternateIdContainer(alternateId)
+      else -> throw IllegalArgumentException("Unknown alternate id: $alternateId")
+    }
+  }
+
+  private fun setAlternateIdContainer(alternateId: Identifier) {
+    alternateIdLabel.visibility = View.VISIBLE
+    alternateIdLabel.text = alternateId.displayType(resources)
+
+    alternateIdContainer.visibility = View.VISIBLE
+
+    inflateAlternateIdView(alternateId.displayValue())
+  }
+
+  private fun inflateAlternateIdView(identifier: String) {
+    val layoutInflater = LayoutInflater.from(context)
+    val alternateIdView = PatientEntryAlternateIdViewBinding.inflate(layoutInflater, this, false)
+    alternateIdView.alternateIdentifier.text = identifier
+    alternateIdContainer.addView(alternateIdView.root)
+  }
+
+  private fun setAlternateIdTextField(alternateId: Identifier) {
+    alternativeIdInputEditText.setTextAndCursor(alternateId.displayValue())
+  }
+
   private fun prefillIdentifier(identifier: Identifier) {
```

Comment:
Don't you think we should add a check here as well to only display identifier if it's a bp passport?

Reply:
We should not be showing nhid or any other id value in bp passport section

Reply:
We will not receive an alternate ID in this field. Hence there is no need to check.

Reply:
yeah, but, it's not that explicit I feel. Here from the code I am not able to make out that this will currently only display the bp passport. For example, even in `EditPatientScreen` we are defining the methods as `displayBPPassport`. I think we should atleast make the method names more explicit.

Reply:
Changing the `prefillIdentifier` function name to `prefillBpPassport`